### PR TITLE
Added a basic layout. Added filter keeping between requests

### PR DESCRIPTION
**Pull request by Alberaan**

### CODE REVIEW

File: alonerpg/tablesreader.py [hunk 1]
```diff
@@ -23,17 +23,17 @@ def __init__(self, index, system, filename):
...
-        return Response(message="Invalid string", data=None)
+        return Response(message="Invalid string", data="")
 
     table_path= tables_path + "/" + chosen_table.system + "/" + table_name + ".txt"
 
     with open(table_path, encoding="utf-8", errors="ignore") as f:
         elements = [line.rstrip() for line in f]
 
-    return Response(message="Result:", data=random.choice(elements))
+    return Response(message="Result for '" + table_name + "':", data=random.choice(elements))
```
Comment (jrclaramunt) [inline alonerpg/tablesreader.py]: Here i would use `f-strings`:
`f"Result for '{table_name}':"`

File: alonerpg/tablesreader.py [hunk 2]
```diff
@@ -51,11 +51,18 @@ def get_tables(path=tables_path):
     return tables_to_return
 
 #lt [filtro]: lista todas las tablas. Si se especifica un filtro, solo se mostrarán aquellas tablas/sistemas cuyo nombre contenga el filtro especificado. Ejemplo: "lt" mostrará todas las tablas. "lt tarot" muestra todas las tablas o sistemas que contengan la palabra tarot
-def lt(tables, filterstring=""):
+def lt(tables, *argv):
     tables_to_return = []
+    
+    if len(argv) == 0:
+        return Response(message="Tablas disponibles:", data=tables)
```
Comment (jrclaramunt) [inline alonerpg/tablesreader.py:58]: Write this in English for language consistency

File: alonerpg/tablesreader.py [hunk 3]
```diff
@@ -51,11 +51,18 @@ def get_tables(path=tables_path):
     return tables_to_return
 
 #lt [filtro]: lista todas las tablas. Si se especifica un filtro, solo se mostrarán aquellas tablas/sistemas cuyo nombre contenga el filtro especificado. Ejemplo: "lt" mostrará todas las tablas. "lt tarot" muestra todas las tablas o sistemas que contengan la palabra tarot
```
Comment (jrclaramunt) [inline alonerpg/tablesreader.py:53]: Write this in English for language consistency

File: alonerpg/tablesreader.py [hunk 4]
```diff
@@ -51,11 +51,18 @@ def get_tables(path=tables_path):
...
     for table in tables:
-        if filterstring.lower() in table.system.lower() or filterstring.lower() in table.filename.lower():
+        flag = True
+        for filterstring in argv:
+            if filterstring.lower() not in table.system.lower() and filterstring.lower() not in table.filename.lower():
+                flag = False
+        if flag:
             tables_to_return.append(table)
 
     return Response(message="Tablas encontradas:", data=tables_to_return)
```
Comment (jrclaramunt) [inline alonerpg/tablesreader.py:68]: Write this in English for language consistency

File: templates/home.html [hunk 5]
```diff
@@ -1,22 +1,54 @@
...
+
+<script>
+function myFunction() {
+    document.getElementById("searchfield").value = "";
+    document.getElementById("filter").submit();
+}
+</script>
 
-</body>
+</body>
```
Comment (jrclaramunt) [inline templates/home.html]: Extra line missing

File: alonerpg/views.py [hunk 6]
```diff
@@ -8,19 +8,30 @@ class HomePageView(TemplateView):
 
     def get(self, request, *args, **kwargs):
         context = self.get_context_data(**kwargs)
-        if "command" in request.GET:
-            command = request.GET["command"]
+        rolled_table = ""
+
+        if "rtn" in request.GET:
+            command = "rtn " + request.GET["rtn"]
```
Comment (jrclaramunt) [inline alonerpg/views.py:15]: If you decide to go with `f-strings`, we should refactor this as well for code consistency

File: alonerpg/views.py [hunk 7]
```diff
@@ -8,19 +8,30 @@ class HomePageView(TemplateView):
...
+            roll_result = dynamic_call(self.tables, command)
+            if isinstance(roll_result.data, str):
+                context["roll_data"] = roll_result.data
+                context["roll_message"] = roll_result.message
```
Comment (jrclaramunt) [inline alonerpg/views.py:19]: Dictionaries can also be defined like this:
```
context = {
    "roll_data": roll_result.data,
    "roll_message": roll_result.message
}
```

**Review by jrclaramunt — CHANGES_REQUESTED** [review]
Good job! Just some minor comments :)

File: alonerpg/tablesreader.py (hunk 1 above)
Comment (Alberaan) [reply alonerpg/tablesreader.py]: Done

File: templates/home.html [hunk 8]
```diff
@@ -1,22 +1,97 @@
...
+
+    favourites.add(favourite)
+    localStorage.favourites = Array.from(favourites).join(",")
+}
+
+function remove_favourite(favourite){
+    if (localStorage.getItem("favourites") == null){
+        return
+    }
+    var favourites = new Set(localStorage.favourites.split(","));
```
Comment (jrclaramunt) [inline templates/home.html]: We can use `var favourites = load_favourites();` here as well

File: templates/home.html [hunk 9]
```diff
@@ -1,22 +1,97 @@
...
+        add_favourite(favourite.split(" ")[1]);
+    } else {
+        remove_favourite(favourite.split(" ")[1]);
+    }
+}
+
+function load_favourites(){
+    if (!localStorage.favourites){
+        return new Set();
+    } else {
```
Comment (jrclaramunt) [inline templates/home.html]: If you still want to get rid of `else`, you can just write:
```
if (!localStorage.favourites){
    return new Set();
}
return new Set(localStorage.favourites.split(","));
```

**Review by jrclaramunt — CHANGES_REQUESTED** [review]

**Review by jrclaramunt — APPROVED** [review]
🏅

**Pull request merged**